tests/test_particlefile.py

```diff
@@ -306,24 +306,27 @@ def SampleP(particle, fieldset, time):
         if time > 5*3600:
             tmp = fieldset.P[particle]  # noqa
 
-    pset = ParticleSet(fieldset, pclass=XiYiParticle, lon=[0], lat=[0.2], lonlatdepth_dtype=np.float64)
+    pset = ParticleSet(fieldset, pclass=XiYiParticle, lon=[0, 0.2], lat=[0.2, 1], lonlatdepth_dtype=np.float64)
     pfile = pset.ParticleFile(name=outfilepath, outputdt=dt)
-    pset.execute([Get_XiYi, SampleP, AdvectionRK4], endtime=10*dt, dt=dt, output_file=pfile)
+    pset.execute([SampleP, Get_XiYi, AdvectionRK4], endtime=10*dt, dt=dt, output_file=pfile)
```

Comment:
The role of SampleP is opaque to me. Can't find a mention of `tmp` in `kernel.py` file.